fabric-biomes-v1/src/main/java/net/fabricmc/fabric/impl/biomes/BaseBiomeEntry.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.biomes;
+
+import net.fabricmc.fabric.api.biomes.v1.OverworldClimate;
+import net.minecraft.util.registry.Registry;
+import net.minecraft.world.biome.Biome;
+
+/**
+ * Represents a biome and its corresponding weight
+ */
+public class BaseBiomeEntry {
+
+	private final Biome biome;
+	private final double weight;
+	private final double upperWeightBound;
+
+	/**
+	 * @param biome the biome
+	 * @param weight how often a biome will be chosen. Most vanilla biomes are a frequency of 1.
+	 * @param climate the climate of the biome entry, just used to store weights
+	 */
+	public BaseBiomeEntry(final Biome biome, final double weight, OverworldClimate climate) {
+		this.biome = biome;
+		this.weight = weight;
+		InternalBiomeData.OVERWORLD_MODDED_BASE_BIOME_WEIGHT_TOTALS.computeIfPresent(climate, (mapClimate, mapWeight) -> mapWeight + weight);
+		InternalBiomeData.OVERWORLD_MODDED_BASE_BIOME_WEIGHT_TOTALS.putIfAbsent(climate, weight);
```

Comment:
The `computeIfPresent` and `putIfAbsent` could be simplified to `compute(climate, (c, w) -> w == null ? weight : weight + w);`

Reply:
I believe prospector has done this